### PR TITLE
fix(ReactTags): Drag and drop failing

### PR DESCRIPTION
**Pull request by cr0ax**
Resurrected from another PR

Hopefully fixes issue #559

I commented on [this PR](https://github.com/prakhar1989/react-tags/pull/583/files) and the author quickly closed it

I will reiterate the same points I stated over on that one, and test it out to see if it fixes the issue

### CODE REVIEW

File: src/components/ReactTags.js [hunk 1]
```diff
@@ -387,7 +387,7 @@ class ReactTags extends Component {
     return tags.map((tag, index) => {
       return (
         <Tag
-          key={`${tag.id}-${index}`}
+          key={tag.key || tag.id}
```
Comment (cr0ax) [inline src/components/ReactTags.js:390]: I don't believe we should have `tag.id` higher priority than `tag.key` for the following reasons:
1) most users this library are not sending a `key` property anyway (virtually no change for them)
2) react uses the `key` property to determine uniqueness already, so this is more in line with the "react way" of doing things

File: src/components/Tag.js [hunk 2]
```diff
@@ -54,6 +54,7 @@ Tag.propTypes = {
   tag: PropTypes.shape({
     id: PropTypes.string.isRequired,
     className: PropTypes.string,
+    key: PropTypes.string,
```
Comment (cr0ax) [inline src/components/Tag.js:57]: this is already optional, omitting it will just default to (something like) the old behavior

**Comment by cr0ax** [comment]
Confirmed this does fix the bug

**Comment by ad1992** [comment]
Thanks for the contribution @cr0ax.

**Pull request merged**